xapi/xapi.go

```diff
@@ -64,24 +64,47 @@ type selectResult struct {
 	aggregate bool
 	fields    []*types.FieldType
 	resp      kv.Response
+
+	results chan PartialResult
+	done    chan error
+}
+
+func (r *selectResult) fetch() {
+	defer close(r.results)
+	for {
+		reader, err := r.resp.Next()
```

Comment:
There should be a way to stop fetching in the middle when `selectResult` is closed by caller.